rpc/grpc/core/src/convert/message.rs

```diff
@@ -474,7 +474,10 @@ try_from!(&protowire::NotifyNewBlockTemplateResponseMessage, RpcResult<kaspa_rpc
 
 try_from!(&protowire::GetCurrentNetworkRequestMessage, kaspa_rpc_core::GetCurrentNetworkRequest);
 try_from!(item: &protowire::GetCurrentNetworkResponseMessage, RpcResult<kaspa_rpc_core::GetCurrentNetworkResponse>, {
-    Self { network: RpcNetworkType::from_str(&item.current_network)? }
+    // Note that current_network is first converted to lowercase because the golang implementation
+    // returns a "human readable" version with a capital first letter while the rusty version
+    // is fully lowercase.
+    Self { network: RpcNetworkType::from_str(&item.current_network.to_lowercase())? }
```

Comment:
Maybe we should call `to_lowercase` within `from_str` to cover more future cases?

Reply:
if you move it to `from_str` perhaps just do
```
...
match network_type {
      "mainnet" | "Mainnet" => Ok(NetworkType::Mainnet),
...
```  
to save the String alloc